SQL: Refactor Tableau connector to make use of the connection properties

bpintea opened the pull request:
This refactors the way the connection URL is being built, to make use of
frameworks' ability to construct the URL based on a set of connection
properties.
This comes with the benefit that the URI attributes escaping is taken
care for us.

The PR also removes the URI "manual" logging, which is already done
by the framework, with proper masking of the password.

bpintea committed:
Refactor to make use of the connection propreties

This refactors the way the connection URL is being built, to make use of
frameworks' ability to construct the URL based on a set of connection
propreties.
This also ensures that the URI attributes are properly escaped.

elasticmachine commented:
Pinging @elastic/es-ql (Team:QL)

bpintea commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into enh/tableau_conn_props

bpintea commented:
@elasticmachine run elasticsearch-ci/bwc

astefan commented on x-pack/plugin/sql/connectors/tableau/connector/connectionProperties.js:
```diff
@@ -0,0 +1,24 @@
+        for (var i = 0; i < avps.length; i++) {
+            var tokens = avps[i].split("=");
+            if (tokens.length != 2 || tokens[0].length == 0 || tokens[1].length == 0) {
+                logging.log("Malformed AVP: [" + avps[i] + "] in additional params: [" + extraProps + "]");
```
I guess what AVP means, but I am not sure. Especially this being a logged message, I'd give it a meaningful name. Something like "Malformed attribute-value pair:......"

bpintea replied:
Acronym expanded.

astefan commented on x-pack/plugin/sql/connectors/tableau/connector/connectionBuilder.js:
```diff
@@ -6,35 +6,7 @@
         urlBuilder += "http://";
     }
-    urlBuilder += attr["server"] + ":" + attr["port"];
+    urlBuilder += attr["server"] + ":" + attr["port"] + "?";
```
Is the question mark at the end always necessary?

bpintea replied:
Yes: the "user" is a mandatory config option (can't submit the form if not filled), so there will always be a property set; so adding the `?` is safe.

astefan reviewed (approved):
LGTM. I've left two comments, though

bpintea committed:
Address review feedback

Rephrase log message.

bpintea committed:
Merge branch 'enh/tableau_conn_props' of github.com:bpintea/elasticsearch into enh/tableau_conn_props

bpintea commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into enh/tableau_conn_props

bpintea committed:
Remove all logging

Remove logging to prevent sensitive info being logged, inline with
Tableau's best practices recommendations.

bpintea commented:
@elasticmachine run elasticsearch-ci/bwc

matriv commented on x-pack/plugin/sql/connectors/tableau/connector/connectionProperties.js:
```diff
@@ -0,0 +1,23 @@
+    if (extraProps != null && extraProps.trim().length > 0) {
+        var avps = extraProps.trim().split(/[\s&]/);
+        for (var i = 0; i < avps.length; i++) {
+            var tokens = avps[i].split("=");
```
Question, to be more lenient shouldn't we trim the split tokens, to treat something like:
` param1  =   value1  `

bpintea replied:
Space is accepted as a delimiter between the AVPs, as a friendlier alternative to `&`; i.e. both forms `a=b&c=d` and `a=b c=d` are permitted. Spaces would normally need to be encoded in a URI.
That's why the initial AVP splitting is done by `.split(/[\s&]/)`.
I've added a comment.

matriv replied:
Thx for explaining!

bpintea commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into enh/tableau_conn_props

bpintea committed:
Add comment on AVPs delimiters

- both `&` and spaces are allowed.

matriv reviewed (approved):
LGTM
